[MNG-4922] ExecutionEvent give on the exception encountered (when having mojoFailed) .
Issue id: MNG-4922


git-svn-id: https://svn.apache.org/repos/asf/maven/maven-3/trunk@1042189 13f79535-47bb-0310-9956-ffa450edef68

+    
+    /**
+     * @since 3.0.2
+     */
+    private Exception exception;
+    /**
+     * @since 3.0.2
+     * @return {@link Exception} encountered during the execution
+     */
+    public Exception getException()
+    {
+        return exception;
+    }
+
+    /**
+     * @since 3.0.2
+     * @param exception
+     */
+    public MojoExecution setException( Exception exception )
+    {
+        this.exception = exception;
+        return this;
+    }
+
